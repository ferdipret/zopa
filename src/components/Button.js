import styled from 'styled-components'
import shadow from './style-utils/shadows'
import color from './style-utils/colors'
import media from './style-utils/media'

export default styled.button`
  background-color: ${color.primary()};
  box-shadow: ${shadow.button.default};
  border: none;
  border-radius: 24px;
  color: white;
  cursor: pointer;
  font-size: 16px;
  height: 48px;
  margin: 60px 0 40px;
  transition: box-shadow 250ms cubic-bezier(0.23, 1, 0.32, 1) 0ms,
    background-color 250ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;
  width: 100%;

  ${media.desktop`
    margin-top: 189px;
  `};

  &:hover {
    background-color: ${color.primary(-3)};
    box-shadow: ${shadow.button.hover};
  }

  &:active {
    background-color: ${color.primary(-6)};
    box-shadow: ${shadow.button.active};
  }
`
