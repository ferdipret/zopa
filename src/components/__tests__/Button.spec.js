import React from "react"
import { shallow } from "enzyme"
import Button from "../Button"
import "jest-styled-components"

describe("Button", () => {
  const onClick = jest.fn()
  const props = {
    onClick
  }
  const wrapper = shallow(<Button {...props} />)

  it("renders properly", () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe("when clicking the button", () => {
    it("calls the `onClick` event", () => {
      wrapper.find("button").simulate("click")
      expect(onClick).toHaveBeenCalled()
    })
  })
})
