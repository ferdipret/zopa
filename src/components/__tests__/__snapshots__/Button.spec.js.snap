// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <styled.button
    onClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "sc-bdVaJa ciFBRb",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "sc-bdVaJa ciFBRb",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
